HW2/P3/mandelbrot.pyx

```diff
@@ -7,65 +8,61 @@ from cython.parallel import prange
+										in_coordsReal[i,j+1],in_coordsReal[i,j])
+					
+				cimag = AVX.make_float8(in_coordsImag[i,j+7],in_coordsImag[i,j+6],
+										in_coordsImag[i,j+5],in_coordsImag[i,j+4],
+										in_coordsImag[i,j+3],in_coordsImag[i,j+2],
+										in_coordsImag[i,j+1],in_coordsImag[i,j])
 
-       # To declare AVX.float8 variables, use:
-       # cdef:
-       #     AVX.float8 v1, v2, v3
-       #
-       # And then, for example, to multiply them
-       #     v3 = AVX.mul(v1, v2)
-       #
-       # You may find the numpy.real() and numpy.imag() fuctions helpful.
-
-    assert in_coords.shape[1] % 8 == 0, "Input array must have 8N columns"
-    assert in_coords.shape[0] == out_counts.shape[0], "Input and output arrays must be the same size"
-    assert in_coords.shape[1] == out_counts.shape[1],  "Input and output arrays must be the same size"
-
-    with nogil:
-        for i in range(in_coords.shape[0]):
-            for j in range(in_coords.shape[1]):
-                c = in_coords[i, j]
-                z = 0
-                for iter in range(max_iterations):
-                    if magnitude_squared(z) > 4:
-                        break
-                    z = z * z + c
-                out_counts[i, j] = iter
-
-
-
-# An example using AVX instructions
-cpdef example_sqrt_8(np.float32_t[:] values):
-    cdef:
-        AVX.float8 avxval
-        float out_vals[8]
-        float [:] out_view = out_vals
-
-    assert values.shape[0] == 8
-
-    # Note that the order of the arguments here is opposite the direction when
-    # we retrieve them into memory.
-    avxval = AVX.make_float8(values[7],
-                             values[6],
-                             values[5],
-                             values[4],
-                             values[3],
-                             values[2],
-                             values[1],
-                             values[0])
-
-    avxval = AVX.sqrt(avxval)
-
-    AVX.to_mem(avxval, &(out_vals[0]))
-
-    return np.array(out_view)
+				toWrite = AVX.float_to_float8(0.0)
+				zreal = AVX.float_to_float8(0.0)
+				zimag = AVX.float_to_float8(0.0)	
+				for iter in xrange(max_iterations):
+				#Calculate magnitude
+					zrealsqr = AVX.mul(zreal,zreal)
+					zimagsqr = AVX.mul(zimag,zimag)
+					mag =AVX.add(zrealsqr,zimagsqr)
+					mask = AVX.greater_than(four, mag)
+				#Break if mag < 4 (in the case that it returns all 0s then all magnitudes are >= 4 thus their sign will return 0 and we should break)
+					sign = AVX.signs(mask)
+					if sign == 0:
+						break
+				#Add 1 to toWrite for those mag < 4
+					addToIter = AVX.bitwise_and(one,mask)
+					toWrite = AVX.add(toWrite,addToIter)
+				#Calculate z= z*z + c
+					zrealtmp = AVX.add(AVX.sub(zrealsqr,zimagsqr),creal)
+					zimagtmp = AVX.fmadd(two,AVX.mul(zreal,zimag),cimag)
+					zreal = zrealtmp
+					zimag = zimagtmp
+			#Write results back
+				AVX.to_mem(toWrite, &(out_counts[i,j]))	
```

Comment:
I think you needed to write a helper function to avoid collisions across threads in assigning values to some temp variable, then iterate through to assign the appropriate values to out_counts (or fix the memory structure, which I don't think I saw).